src/UserFunctions.java

```diff
@@ -51,8 +55,8 @@ public bool parseCommand(String input) {
                 printCommands();
                 break;
             case NEW_COMMAND:
-                if (inputs.length > 1)
-                    createNewPuzzle(inputs[1]);
+                if (parameter != null && !parameter.equals(""))
+                    createNewPuzzle(parameter);
                 else
                     createNewPuzzle("");
```

Comment:
This is another place I would use braces to make sure that the if-else flow is correct even if we add code to one of them.

```suggestion
                if (parameter != null && !parameter.equals("")) {
                    createNewPuzzle(parameter);
                } else {
                    createNewPuzzle("");
                }
```